src/node_report.cc

```diff
@@ -0,0 +1,921 @@
+time_t boot_time;  // Node.js boot time absolute
+
+// External function to trigger a node report, writing to file.
+// The 'name' parameter is in/out: an input filename is used
+// if supplied, and the actual filename is returned.
+void TriggerNodeReport(Isolate* isolate,
+                       Environment* env,
+                       const char* message,
+                       const char* location,
+                       std::string name,
+                       Local<String> stackstr) {
+  std::ostringstream oss;
+  std::string filename;
+  std::shared_ptr<PerIsolateOptions> options;
+  if (env != nullptr) options = env->isolate_data()->options();
+
+  // Obtain the current time and the pid (platform dependent)
+  TIME_TYPE tm_struct;
+  PID_TYPE pid;
+  LocalTime(&tm_struct);
+  pid = uv_os_getpid();
+  // Determine the required report filename. In order of priority:
+  //   1) supplied on API 2) configured on startup 3) default generated
+  if (!name.empty()) {
+    // Filename was specified as API parameter, use that
+    oss << name;
+  } else if (env != nullptr && options->report_filename.length() > 0) {
+    // File name was supplied via start-up option, use that
+    oss << options->report_filename;
+  } else {
+    // Construct the report filename, with timestamp, pid and sequence number
+    oss << "node-report";
+    seq++;
+#ifdef _WIN32
+    oss << "." << std::setfill('0') << std::setw(4) << tm_struct.wYear;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMonth;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wDay;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.wHour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wMinute;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.wSecond;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#else  // UNIX, OSX
+    oss << "." << std::setfill('0') << std::setw(4) << tm_struct.tm_year+1900;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mon + 1;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_mday;
+    oss << "." << std::setfill('0') << std::setw(2) << tm_struct.tm_hour;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_min;
+    oss << std::setfill('0') << std::setw(2) << tm_struct.tm_sec;
+    oss << "." << pid;
+    oss << "." << std::setfill('0') << std::setw(3) << seq.load();
+#endif
+    oss << ".json";
+  }
+
+  filename = oss.str();
+  // Open the report file stream for writing. Supports stdout/err,
+  // user-specified or (default) generated name
+  std::ofstream outfile;
+  std::ostream* outstream = &std::cout;
+  if (filename == "stdout") {
+    outstream = &std::cout;
+  } else if (filename == "stderr") {
+    outstream = &std::cerr;
+  } else {
+    // Regular file. Append filename to directory path if one was specified
+    if (env != nullptr && options->report_directory.length() > 0) {
```

Comment:
This doesn’t work if `filename` is an absolute path, e.g. when called as `report.triggerReport('/tmp/report.json')` – in that case we’d want to also take the `else` case.

Reply:
Providing a response somewhere down; where you mentioned similar comment in the doc.